sdks/java/core/src/main/java/org/apache/beam/sdk/testing/PAssert.java

```diff
@@ -1035,15 +1101,8 @@ private SideInputCheckerDoFn(
 
     @ProcessElement
     public void processElement(ProcessContext c) {
-      try {
```

Comment:
This suppression of exceptions was the reason the side input tests failed in Flink runner running in streaming mode.
Removing this solves the issue. All the ROS tests pass now (Including the new ones).
@kennknowles PTAL